Framework/API/inc/MantidAPI/MatrixWorkspace.h

```diff
@@ -147,7 +147,7 @@ class MANTID_API_DLL MatrixWorkspace : public IMDWorkspace,
   Types::Core::DateAndTime getLastPulseTime() const;
 
   /// Returns the y index which corresponds to the X Value provided
-  std::size_t yIndexOfX(const double xValue, const std::size_t = 0,
+  std::size_t yIndexOfX(const double xValue, const std::size_t &index = 0,
```

Comment:
One more question on this - why did you change the index parameter here to be passed by reference? In various other places you changed some other parameters that had primitive types to be passed by value. That seemed sensible since I don't think there's any upside to passing them by const ref. But this one went in the other direction.

Reply:
Your right, I don't think there is any benefit to passing `std::size_t` by reference as it is a primitive type. However, I like the consistency of passing all standard library data types (e.g. `std::string` , `std::vector` , `std::size_t`) by reference. From a stylistic viewpoint it just seems better (maybe only to me 😄)

Reply:
OK, seems like you're not alone looking elsewhere in the Mantid code eg
```
NumericAxis::NumericAxis(const std::size_t &length)
```
Fine to go with this